test/suites/app-service-configurable/config_test.go

```diff
@@ -0,0 +1,80 @@
+package test
+
+import (
+	"edgex-snap-testing/test/utils"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Deprecated
+func TestEnvConfig(t *testing.T) {
+	// start clean
+	utils.SnapStop(t, ascService)
+
+	t.Run("change service port", func(t *testing.T) {
+		t.Cleanup(func() {
+			utils.SnapStop(t, ascService)
+			utils.SnapUnset(t, ascSnap, "env.service.port")
+		})
+
+		const newPort = "56789"
+
+		// make sure the port is available before using it
+		utils.CheckPortAvailable(t, newPort)
+
+		utils.SnapSet(t, ascSnap, "env.service.port", newPort)
+		utils.SnapStart(t, ascSnap)
+		utils.WaitServiceOnline(t, newPort)
+	})
+
+	t.Run("set profile", func(t *testing.T) {
+		t.Cleanup(func() {
+			utils.SnapStop(t, ascService)
+		})
+
+		time := time.Now()
+		profile := "http-export"
+
+		// set profile
+		utils.SnapSet(t, ascSnap, "profile", profile)
+		utils.SnapStart(t, ascSnap)
+
+		//check logs for the record of expected profile
+		time.Sleep(1 * time.Second)
+		logs := utils.SnapLogsJournal(t, time, ascSnap)
+		expectLog := "app=app-" + profile
+		require.True(t, strings.Contains(logs, expectLog))
+	})
+}
+
+func TestAppConfig(t *testing.T) {
+	t.Skip("TODO")
+}
+
+func TestProfileConfig(t *testing.T) {
+	// start clean
+	utils.SnapStop(t, ascService)
+
+	t.Run("set profile", func(t *testing.T) {
+		t.Cleanup(func() {
+			utils.SnapStop(t, ascService)
+		})
+
+		start := time.Now()
+		profile := "http-export"
+
+		// set profile
+		utils.SnapSet(t, ascSnap, "profile", profile)
```

Comment:
Usually you have to clean this up by unsetting, but because of the asc service wrapper bug, there is no valid fallback profile. The default in tests is rules-engine. Maybe add a constant in main_test as `defaultProfile=rules-engine` and use it here to cleanup and also in main to initialize.